app/src/behaviors/behavior_sensor_rotate.c

```diff
@@ -0,0 +1,99 @@
+#include <logging/log.h>
+
+#include <drivers/sensor.h>
+#include <zmk/event_manager.h>
+#include <zmk/events/keycode_state_changed.h>
+#include <zmk/keymap.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+
+struct behavior_sensor_rotate_config {
+    struct zmk_behavior_binding cw_binding;
+    struct zmk_behavior_binding ccw_binding;
+};
+
+struct behavior_sensor_rotate_data {
+    struct zmk_behavior_binding *triggered_binding;
+};
+
+static int on_sensor_binding_triggered(struct zmk_behavior_binding *binding,
+                                       const struct device *sensor, int64_t timestamp) {
+    const struct device *dev = device_get_binding(binding->behavior_dev);
+    const struct behavior_sensor_rotate_config *cfg = dev->config;
+    struct behavior_sensor_rotate_data *data = dev->data;
+
+    struct sensor_value value;
+    int err;
+
+    err = sensor_channel_get(sensor, SENSOR_CHAN_ROTATION, &value);
+
+    if (err) {
+        LOG_WRN("Failed to get sensor rotation value: %d", err);
+        return err;
+    }
+
+    switch (value.val1) {
+    case 1:
+        data->triggered_binding = (struct zmk_behavior_binding *)&cfg->cw_binding;
+        break;
+    case -1:
+        data->triggered_binding = (struct zmk_behavior_binding *)&cfg->ccw_binding;
+        break;
+    default:
+        return -ENOTSUP;
+    }
+
+    LOG_DBG("Sensor binding: %s", log_strdup(binding->behavior_dev));
+
+    struct zmk_behavior_binding_event event = {.layer = zmk_keymap_highest_layer_active(),
+                                               .timestamp = timestamp};
+
+    behavior_keymap_binding_pressed(data->triggered_binding, event);
+    k_msleep(5);
+    behavior_keymap_binding_released(data->triggered_binding, event);
+
+    return ZMK_BEHAVIOR_OPAQUE;
+
+    // return behavior_sensor_keymap_binding_triggered(data->triggered_binding, sensor, timestamp);
+}
+
+static const struct behavior_driver_api behavior_sensor_rotate_driver_api = {
+    .sensor_binding_triggered = on_sensor_binding_triggered};
+
+static int behavior_sensor_rotate_init(const struct device *dev) { return 0; };
+
+#define _TRANSFORM_ENTRY(idx, node)                                                                \
+    {                                                                                              \
+        .behavior_dev = DT_LABEL(DT_INST_PHANDLE_BY_IDX(node, bindings, idx)),                     \
+        .param1 = COND_CODE_0(DT_INST_PHA_HAS_CELL_AT_IDX(node, bindings, idx, param1), (0),       \
+                              (DT_INST_PHA_BY_IDX(node, bindings, idx, param1))),                  \
+        .param2 = COND_CODE_0(DT_INST_PHA_HAS_CELL_AT_IDX(node, bindings, idx, param2), (0),       \
+                              (DT_INST_PHA_BY_IDX(node, bindings, idx, param2))),                  \
+    }
+
+#define KP_INST(n)                                                                                 \
```

Comment:
Minor: I'm guessing you copied this from the key press behavior? The naming of this doesn't make a lot of sense here.

Reply:
I think I copied from mod-morph but it seems like most of the behaviors use KP e.g. tap dance.

Reply:
This is something that will be addressed in an upcoming PR. Functionally, it doesn't matter what the `XX_INST(n)` is labelled as, but consistency with the behavior name is nice to have.

Reply:
Changed the name to match the behavior.